airbyte-integrations/connectors/source-exchange-rates/source_exchange_rates/source.py

```diff
@@ -0,0 +1,101 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import pendulum
+import requests
+from base_python import AbstractSource, HttpStream, Stream
+from pendulum import DateTime
+
+
+class ExchangeRates(HttpStream):
+    date_field_name = "date"
+
+    # HttpStream related fields
+    url_base = "https://api.ratesapi.io/"
+    cursor_field = date_field_name
+
+    def __init__(self, base: str, start_date: DateTime):
+        super().__init__()
+        self._base = base
+        self._start_date = start_date
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
```

Comment:
no need to override this method, returns None by default

Reply:
I don't think that's true? https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/base-python/base_python/sdk/streams/http.py#L69

should we make this return `None`?

Reply:
oh nvm my mistake! I think I made it abstract to "force" the user to take a stance on pagination. LGTM then